staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -314,6 +363,47 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 					defer proxy.Close()
 				}
 
+				var l net.Listener
+				if testCase.socks5ProxyServerFunc != nil {
+					var proxyHandler *socks5.Server
+					if testCase.proxyAuth != nil {
+						proxyHandler = socks5Server(&socks5.StaticCredentials{
+							"proxyuser": "proxypasswd",
+						})
+						proxyCalledWithAuth = true
```

Comment:
this seems misleading... this is setting proxyCalledWithAuth when the proxy server hasn't even been called yet. compare with the way http proxy sets this inside the server handler based on the incoming request

Reply:
setting a CredentialStore implementation that logs an auth attempt would match the http test better and would let us ensure the socks server was called, and that it was called with auth, and the content of the auth credentials